examples/marquee.lisp

```diff
@@ -0,0 +1,89 @@
+(defpackage charms-marquee
+  (:use :cl)
+  (:export :main))
+(in-package :charms-marquee)
+
+(defclass marquee ()
+  ((position :initarg :position :accessor marquee-position)
+   (size :initarg :size :accessor marquee-size)
+   (text :initarg :text :accessor marquee-text)
+   (viewport :initarg :viewport :reader marquee-viewport)
+   (padding-char :initarg :padding-char :initform #\Space :accessor marquee-padding-char)
+   (buffer :initform "" :accessor marquee-buffer))
+  (:documentation "A marquee is like a scrolling text thing. A string of characters moves through
+a viewport. The viewport constrains the text so you only see a little bit at a time."))
+
+(defmethod marquee-viewport :before ((marquee marquee))
```

Comment:
any reason to use methods here and below? is the :BEFORE needed from a protocol sense?

Reply:
That can go. I was just fooling around with some other way of doing this and forgot to clean it up.